save-cosv/src/main/kotlin/com/saveourtool/save/cosv/cvsscalculator/ScoreCalculator.kt

```diff
@@ -0,0 +1,98 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.cosv.cvsscalculator
+
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.ATTACK_COMPLEXITY
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.ATTACK_VECTOR
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.AVAILABILITY
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.CONFIDENTIALITY
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.CVSS_VERSION
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.INTEGRITY
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.PRIVILEGES_REQUIRED
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.SCOPE
+import com.saveourtool.save.cosv.cvsscalculator.CvssMetrics.Companion.USER_INTERACTION
+import java.lang.Float.min
+import kotlin.math.pow
+import kotlin.math.roundToInt
+
+private class CvssMetrics {
+    companion object {
+        const val ATTACK_COMPLEXITY = "/AC:"
+        const val ATTACK_VECTOR = "/AV:"
+        const val AVAILABILITY = "/A:"
+        const val CONFIDENTIALITY = "/C:"
+        const val CVSS_VERSION = "CVSS:"
+        const val INTEGRITY = "/I:"
+        const val PRIVILEGES_REQUIRED = "/PR:"
+        const val SCOPE = "/S:"
+        const val USER_INTERACTION = "/UI:"
+    }
+}
+
+@Suppress(
+    "UnsafeCallOnNullableType",
```

Comment:
maybe add useful exception here instead of `!!`?

Reply:
Added